calico/felix/ipsets.py

```diff
@@ -64,8 +63,11 @@ def __init__(self, ip_type):
         self.tags_by_prof_id = {}
         self.endpoints_by_ep_id = {}
 
-        # Indexes.
-        self.endpoint_ids_by_tag = defaultdict(set)
+        # Main index self.ip_owners_by_tag[tag][ip] == set([endpoint_id])
```

Comment:
I found this comment pretty confusing. The point of this ip_owners_by_tag thing is that when you remove an endpoint, you want to remove its IP addresses from ips_in_tag if another endpoint had the same ID (which is possible, thanks to anycast). I think you need a comment (just a line or two) making this clear.